pages/snake.js

```diff
@@ -71,12 +84,45 @@ export default function Snake() {
         }
     }
 
+    function drawFood(foodX, foodY, snakeboardCTX) {
+        snakeboardCTX.fillStyle = 'lightgreen';
+        snakeboardCTX.strokeStyle = 'darkgreen';
+        snakeboardCTX.fillRect(foodX, foodY, 10, 10);
+        snakeboardCTX.strokeRect(foodX, foodY, 10, 10);
+    }
+
+    function random_food(min, max) {
+        return Math.round((Math.random() * (max - min) + min) / 10) * 10;
+    }
+
+    function generate_food(snakeboard, snake) {
+        foodX = random_food(0, snakeboard.width - 10);
+        foodY = random_food(0, snakeboard.height - 10);
+        snake.forEach(function has_snake_eaten_food(part) {
+            const has_eaten = part.x === foodX && part.y === foodY;
+            if (has_eaten) generate_food();
+        });
+    }
+
+    function hasGameEnded(snake, snakeboard) {
```

Comment:
I think it's not common to name functions as questions. Maybe name it `endGame` ?